distage/distage-testkit/src/main/scala/com/github/pshirshov/izumi/distage/testkit/TestComponentsLifecycleManager.scala

```diff
@@ -0,0 +1,39 @@
+package com.github.pshirshov.izumi.distage.testkit
+
+import com.github.pshirshov.izumi.distage.roles.launcher.{ComponentLifecycle, ComponentsLifecycleManagerImpl}
+import com.github.pshirshov.izumi.distage.roles.roles.RoleComponent
+import com.github.pshirshov.izumi.distage.testkit.MemoizingDistageResourceCollection.memoizedComponentsLifecycle
```

Comment:
You probably still want to add a method in ResourceCollection instead of directly accessing mutable globals. You might want to make everything in the `MemoizingDistageResourceCollection` object private.